phylanx/execution_tree/primitives/mod_operation.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017-2018 Hartmut Kaiser
+// Copyright (c) 2018 Shahrzad Shirzad
+// Copyright (c) 2018 Parsa Amini
+// Copyright (c) 2020 Steven R. Brandt
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_PRIMITIVES_MOD_OPERATION_JUN_18_2020_1202PM)
+#define PHYLANX_PRIMITIVES_MOD_OPERATION_JUN_18_2020_1202PM
+
+#include <phylanx/config.hpp>
+
+#include <hpx/futures/future.hpp>
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        struct mod_op;
+    }
+
+    template<typename T>
+    class base
+      : public primitive_component_base
+      , public std::enable_shared_from_this<T>
+    {
+    public:
+        base() = default;
+
+        base(primitive_arguments_type&& operands,
+            std::string const& name, std::string const& codename)
+        : primitive_component_base(std::move(operands), name, codename) {}
+    };
```

Comment:
There is no need to have this base class, I think. Directly deriving the `mod_operation` primitive below from both, `primitive_component_base> and `std::enable_shared_from_this<mod_operation>` should be sufficient.

Reply:
@hkaiser  need something for the base_type field, and it has to somehow include both primitive_component_base and std::shared_from_this<mod_operation>.

Reply:
I added a suggestion below, HTH.